File: 后端/src/main/java/org/xmexg/wkexpress/tools/ResponseModel.java
package org.xmexg.wkexpress.tools;
import com.alibaba.fastjson.JSON;
import lombok.Getter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

/**
 * 封装返回结果
 */
@Getter
@Component
public class ResponseModel {
    public static final int SUCCESS = 200;
    public static final int FAIL = 400;
    public static final int UNAUTHORIZED = 401;
    public static final int invaliduser = 403;
    public static final int NOTFOUND = 404;
    public static final int notlogin = 405;

    private int code;
    private String message;
    private Object data;

    public String make(int code, String message, Object data) {
        this.code = code;
        this.message = message;
        this.data = data==null?"":data;
        return JSON.toJSONString(this);
    }

    public String success() {
        return make(SUCCESS, "success", null);
    }

    public String success(Object data) {
        return make(SUCCESS, "success", data);
    }

    public String fail() {
        return make(FAIL, "fail", null);
    }

    public String fail(String message) {
        return make(FAIL, message, null);
    }

    public String unauthorized() {
        return make(UNAUTHORIZED, "unauthorized", null);
    }

    public String unauthorized(String message) {
        return make(UNAUTHORIZED, message, null);
    }

    public String invaliduser() {
        return make(invaliduser, "invaliduser", null);
    }

    public String invaliduser(String message) {
        return make(invaliduser, message, null);
    }

    public String notfound() {
        return make(NOTFOUND, "notfound", null);
    }

    public String notfound(String message) {
        return make(NOTFOUND, message, null);
    }

    public String notlogin() {
        return make(notlogin, "notlogin", null);
    }

    public String notlogin(String message) {
        return make(notlogin, message, null);
    }

    @Override
    public String toString() {
        return JSON.toJSONString(this);
    }

}

File: 扩展web管理/wkexpress_web/src/main/kotlin/org/looom/wkexpress_web/util/ResponseModel.kt
package org.looom.wkexpress_web.util

import com.alibaba.fastjson2.JSON

/**
 * 封装返回结果
 */
class ResponseModel {

    // 创建伴生对象
    companion object {

        var code = 0
        var message: String? = null
        var data: Any? = null

        @JvmStatic  // 使static
        fun make(code: Int, message: String?, data: Any?): String {
            this.code = code
            this.message = message
            this.data = data ?: ""
            return JSON.toJSONString(this)
        }

        @JvmStatic
        fun success(): String {
            return make(SUCCESS, "success", null)
        }

        @JvmStatic
        fun success(data: Any?): String {
            return make(SUCCESS, "success", data)
        }

        @JvmStatic
        fun fail(): String {
            return make(FAIL, "fail", null)
        }

        @JvmStatic
        fun fail(message: String?): String {
            return make(FAIL, message, null)
        }

        @JvmStatic
        fun unauthorized(): String {
            return make(UNAUTHORIZED, "用户未认证", null)
        }

        @JvmStatic
        fun unauthorized(message: String?): String {
            return make(UNAUTHORIZED, message, null)
        }

        @JvmStatic
        fun invaliduser(): String {
            return make(INVALIDUSER, "无效用户", null)
        }

        @JvmStatic
        fun invaliduser(message: String?): String {
            return make(INVALIDUSER, message, null)
        }

        @JvmStatic
        fun notfound(): String {
            return make(NOTFOUND, "未找到", null)
        }

        @JvmStatic
        fun notfound(message: String?): String {
            return make(NOTFOUND, message, null)
        }

        @JvmStatic
        fun notlogin(): String {
            return make(NOTLOGIN, "用户未登录", null)
        }

        @JvmStatic
        fun notlogin(message: String?): String {
            return make(NOTLOGIN, message, null)
        }

        override fun toString(): String {
            return JSON.toJSONString(this)
        }

        const val SUCCESS: Int = 200
        const val FAIL: Int = 400
        const val UNAUTHORIZED: Int = 401
        const val INVALIDUSER: Int = 403
        const val NOTFOUND: Int = 404
        const val NOTLOGIN: Int = 405
    }
}